x-pack/plugins/fleet/server/services/agents/crud.test.ts

```diff
@@ -281,8 +288,136 @@ describe('Agents CRUD test', () => {
       });
     });
 
+    it('should return correct status summary when showUpgradeable is selected and total is less than limit', async () => {
+      searchMock.mockImplementationOnce(() =>
+        Promise.resolve(getEsResponse(['1', '2', '3', 'up', '5'], 100, 'updating'))
+      );
+      searchMock.mockImplementationOnce(() =>
+        Promise.resolve(getEsResponse(['1', '2', '3', 'up', '5'], 100, 'updating'))
+      );
+      const result = await getAgentsByKuery(esClientMock, soClientMock, {
+        showUpgradeable: true,
+        showInactive: false,
+        getStatusSummary: true,
+        page: 1,
+        perPage: 5,
+      });
+
+      expect(result).toEqual({
+        agents: [
```

Comment:
nit: is it necessary to verify the complete result? You can use `expect.objectContaining` to only include fields which are relevant to the test.